src/Microsoft.TemplateEngine.Authoring.TemplateVerifier/VerificationEngine.cs

```diff
@@ -0,0 +1,335 @@
+            {
+                return options.CustomDirectoryVerifier(
+                    contentDir,
+                    new Lazy<IAsyncEnumerable<(string FilePath, string ScrubbedContent)>>(
+                        GetVerificationContent(contentDir, globs, options.CustomScrubbers)));
+            }
+
+            Verifier.DerivePathInfo(
+                (sourceFile, projectDirectory, type, method) => new(
+                    directory: options.ExpectationsDirectory ?? "VerifyExpectations",
+                    typeName: options.TemplateName,
+                    methodName: EncodeArgsAsPath(options.TemplateSpecificArgs)));
+
+            if ((options.UniqueFor ?? UniqueForOption.None) != UniqueForOption.None)
+            {
+                foreach (UniqueForOption value in Enum.GetValues(typeof(UniqueForOption)))
+                {
+                    if ((options.UniqueFor & value) == value)
+                    {
+                        switch (value)
+                        {
+                            case UniqueForOption.None:
+                                break;
+                            case UniqueForOption.Architecture:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForArchitecture();
+                                break;
+                            case UniqueForOption.OsPlatform:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForOSPlatform();
+                                break;
+                            case UniqueForOption.Runtime:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForRuntime();
+                                break;
+                            case UniqueForOption.RuntimeAndVersion:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForRuntimeAndVersion();
+                                break;
+                            case UniqueForOption.TargetFramework:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForTargetFramework();
+                                break;
+                            case UniqueForOption.TargetFrameworkAndVersion:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForTargetFrameworkAndVersion();
+                                break;
+                            default:
+                                throw new ArgumentOutOfRangeException();
+                        }
+                    }
+                }
+            }
+
+            if (options.DisableDiffTool ?? false)
+            {
+                defaultVerifyTask = defaultVerifyTask.DisableDiff();
+            }
+
+            return defaultVerifyTask;
+        }
+
+        private async Task VerifyResult(TemplateVerifierOptions args, CommandResult commandResult)
+        {
+            // Customize diff output of verifier
+            VerifyDiffPlex.Initialize(OutputType.Compact);
```

Comment:
this api is meant to be called once at startup. usually in a `[ModuleInitializer]`